tests/Formatter/SmartJsonFormatterTest.php

```diff
@@ -95,4 +98,140 @@ public function test_format_ignore_paths(): void
+                        [2 => true],
+                    ],
+                    'type'     => '->',
+                    'file'     => '/vendor/illuminate/database/Connectors/Connector.php',
+                    'line'     => 46,
+                ],
+                [
+                    'class'    => 'Illuminate\\Database\\Connectors\\Connector',
+                    'function' => 'createConnection',
+                    'args'     => [
+                        'mysql:host=test.database.hostname.com;port=3306;dbname=test',
+                        [
+                            'username' => 'username',
+                            'password' => 'password',
+                        ],
+                        [2 => true],
+                    ],
+                    'type'     => '->',
+                    'file'     => '/vendor/illuminate/database/Connectors/MySqlConnector.php',
+                    'line'     => 24,
+                ],
+                [
+                    'function' => 'prepareDatabase',
+                    'args'     => [
+                        'foo',
+                        'bar',
+                        [2 => true],
+                        42,
+                        42.42,
+                        new \stdClass,
+                    ],
+                    'file'     => '/my/fake/path/prepare.php',
+                    'line'     => 42,
+                ],
+                [
+                    'function' => 'connectToDatabase',
+                    'args'     => [],
+                    'file'     => '/my/fake/path/connect.php',
+                    'line'     => 42,
+                ],
+            ]),
+        ]);
+
+        $formatter = new SmartJsonFormatter;
+        $formatter->addMap(new RedactArgumentsMap($redactedFrameArguments));
+        $output = json_decode($formatter->format($record), true);
+
+        $this->assertCount(5, $output['lines'][0]['meta']['exception']['trace']);
+
+        $expectedArguments = [
+            [
+                'string(***REDACTED***)',
+                'string(***REDACTED***)',
+                'string(***REDACTED***)',
+                'array(***REDACTED***)',
+                'int(***REDACTED***)',
+                'float(***REDACTED***)',
+                'stdClass',
+            ],
+            [
+                'string(***REDACTED***)',
+                'string(***REDACTED***)',
+                'string(***REDACTED***)',
+                'array(***REDACTED***)',
+            ],
+            [
+                'string(***REDACTED***)',
+                'array(***REDACTED***)',
+                'array(***REDACTED***)',
+            ],
+            [
+                'string(foo)',
+                'string(bar)',
+                'array(1)',
+                'int(42)',
+                'float(42.42)',
+                'stdClass',
+            ],
+            [],
+        ];
+
+        foreach ($expectedArguments as $index => $expectedArgument) {
```

Comment:
What do you think about splitting this loop out into a data provider?

Reply:
@liamkeily I don't think so. We're test a single stack trace which contains multiple frames.